custom_components/rrd/camera.py

```diff
@@ -72,20 +72,42 @@ def __init__(self, name, rrd, width, height, timerange, args):
         self._defs = []
         self._lines = []
         try:
-            info = rrdtool.info(self._rrd)
-            self._step = info["step"]
-            for key, value in info.items():
+            rrdinfo = rrdtool.info(self._rrd)
+            self._step = rrdinfo["step"]
+            for key, value in rrdinfo.items():
                 if ".index" in key:
-                    ds = re.search(r"\[(.*?)\]", key).group(1)
+                    ds = re.search(r"\[(.*?)\]", key).group(1)  # Datasource name
                     self._unique_id += f"_{ds}"
-                    cf = info[f"rra[{value}].cf"]
-                    self._defs.append(f"DEF:{ds.capitalize()}={self._rrd}:{ds}:{cf}")
+                    cf = rrdinfo[f"rra[0].cf"]
+
+                    # Append DEF of primary DS RRA
+                    graph_def = f"DEF:{ds.capitalize()}={rrd}:{ds}:{cf}"
+                    self._defs.append(graph_def)
+                    _LOGGER.debug('Added graph %s', graph_def)
+
+                    # Append all other RRAs as DEF with names "{ds.capitalize()}{rra_pdp_per_row}"
+                    rra_index = 1
+                    while True:
```

Comment:
Isn't this loop doing the same as the for in line 77 ?

Reply:
Loop on line 77 is around all `ds` items, where indexes are datasource names.
But this loop is around all `rra` items, where indexes are numbers.

See the `rrdinfo` example in comment above. It is visible there, that there are these two types of items.

Reply:
Can we close this thread based on my previous explanation? :-)

Reply:
based on my previous comment, why not iterate `rrdinfo` with `enumerate` ? this way you get the `rra_index`

Reply:
Cannot enumerate easily, because keys are not `0`, `1`, .... Keys are `rra[0].cf`, `rra[0].rows`, `rra[0].cur_row`, `rra[0].pdp_per_row`, ... and similarly with `rra[1].cf`, etc.  .. So for every RRA there are multiple keys.